### DISCUSSION TITLE
createSlice and js class as a CreateSliceOptions.initialState

### DISCUSSION THREAD

**Discussion by dmitrybv** [reduxjs/redux-toolkit#3418] [Q&A]
I wonder if the createSlice function does or does not support classes as a CreateSliceOptions parameter. initialState.

If in the demo project we replace interface CounterState with class CounterState

```typescript
export class CounterState {
   public value: number = 0;
   public status: 'idle' | 'loading' | 'failed' = 'idle';
}

const initialState: CounterState = new CounterState();
```

then no errors are thrown when compiling typescript.

Errors occur while the application is running, but the text of the error does not make it clear whether the error is related to the use of classes.

**Comment by EskiMojo14** [comment]
Putting classes in your Redux store is [not recommended](https://redux.js.org/faq/organizing-state#can-i-put-functions-promises-or-other-non-serializable-items-in-my-store-state).

They're not serialisable, meaning things like persistence and time travel debugging will not work properly. Additionally, classes tend to modify their state **mutably**, whereas Redux relies on state being changed **immutably**.
